hedera-node/src/main/java/com/hedera/services/state/logic/BlockManager.java

```diff
@@ -84,12 +88,65 @@ public long getCurrentBlockNumber() {
 		return networkCtx.get().getManagedBlockNo();
 	}
 
+	public long getProvisionalBlockNumber() {
+		return provisionalBlockNumber;
+	}
+
+	public org.hyperledger.besu.datatypes.Hash getProvisionalBlockHash(long blockNumber) {
+		if (blockNumber == provisionalBlockNumber) {
+			return provisionalBlockHash;
+		}
+		return networkCtx.get().getBlockHashByNumber(blockNumber);
+	}
+
+	/**
+	 *
+	 * @param timestamp
+	 * @param gasLimit
+	 * @return
+	 */
+	public HederaBlockValues createProvisionalBlockValues(@NotNull final Instant timestamp, long gasLimit) {
+		var curNetworkCtx = networkCtx.get();
+
+		var blockNo = curNetworkCtx.getBlockNo();
+
+		// Before the 0.26 upgrade, we used the consensus timestamp as the block number; and if we
+		// get a zero block number, it means the post-0.26 block sync hasn't happened yet
+		Instant provisionalBlockTimestamp;
+		if (blockNo == 0) {
+			provisionalBlockNumber = timestamp.getEpochSecond();
+			provisionalBlockTimestamp = Instant.EPOCH;
+			provisionalBlockHash = UNAVAILABLE_BLOCK_HASH;
+		} else if (willCreateNewBlock(timestamp)) {
```

Comment:
It would be good if we could re-use this `willCreateNewBlock()` method here and in `getManagedBlockNumberAt()`;  would also be nice to re-use the `InterruptedException` block that gets the pending block hash.